packages/cxl-ui/src/components/cxl-marketing-nav.js

```diff
@@ -1,631 +1,339 @@
-        { component: searchFormTemplate.content.firstElementChild },
-      ];
+      menuItemElement.addEventListener('click', this._onBackBtnClick.bind(this));
     }
 
-    /**
-     * Enable instant typing, avoid focus click.
-     */
-    menuItemSearchContextMenu.$.overlay.addEventListener('vaadin-overlay-open', (e) =>
-      e.target.querySelector('#search-input').focus()
-    );
-  }
-
-  /**
-   * @private
-   */
-  _updatedWide() {
-    this._reorientMenuItems();
-    this._rotateMenuItemSearchListenOn();
+    return menuItemElement;
   }
 
-  /**
-   * Mobile menu toggle.
-   *
-   * @see https://app.clickup.com/t/3phu3bv
-   * @since 2022.10.18
-   */
-  // eslint-disable-next-line class-methods-use-this
-  _toggleMobileMenu() {
-    const bodyElement = document.querySelector('body');
-
-    if (bodyElement) {
-      bodyElement.classList.toggle('cxl-mobile-menu-opened');
-    }
-  }
+  _updateContextMenuItems() {
+    if (!this.menuBars) return;
 
-  _menuShadowItemsClick(e) {
-    if (this.menuShadowItemsSelectedIdx !== -1) {
-      if (this.menuShadowItemsSelectedIdx === e.currentTarget.selected) {
-        e.currentTarget.selected = -1;
-        e.stopImmediatePropagation();
-      }
-    }
+    [...this.menuBars].forEach((menu) => {
+      const splitItems = menu.shadowRoot.querySelectorAll(
+        'vaadin-context-menu-item.menu-item-split-nav'
+      );
+      [...splitItems].forEach((item) => {
+        if (!item.parentElement.classList.contains('menu-item-split-nav')) {
+          item.parentElement.classList.add('menu-item-split-nav');
+        }
+      });
+    });
   }
 
-  /**
-   * Maybe close context menu overlay on scroll?
-   * Only applicable if it's shown as dropdown.
-   *
-   * @see https://app.clickup.com/t/3q2z1a3
-   * @since 2022.10.19
-   */
-  _maybeCloseContextMenuOverlay() {
-    const el = document.querySelector('vaadin-context-menu-overlay');
+  _replaceMenuIcon() {
+    if (!this.menuBars || this.wide) return;
 
-    if (el && !this._phone) {
-      el.close();
-    }
+    [...this.menuBars].forEach((menu) => {
+      const overflowMenuButton = menu.shadowRoot.querySelector(
+        'vaadin-menu-bar-button[part="overflow-button"]'
+      );
+      overflowMenuButton.toggleAttribute('hidden', false);
+      if (overflowMenuButton && !overflowMenuButton.iconFixed) {
+        const menuIcon = document.createElement('vaadin-icon');
+        const closeIcon = document.createElement('vaadin-icon');
+        menuIcon.setAttribute('icon', 'lumo:menu');
+        closeIcon.setAttribute('icon', 'lumo:cross');
+        menuIcon.classList.add('vaadin-menu-bar-button--icon');
+        menuIcon.classList.add('menu-icon');
+        closeIcon.classList.add('vaadin-menu-bar-button--icon');
+        closeIcon.classList.add('close-icon');
+        overflowMenuButton.appendChild(closeIcon);
+        overflowMenuButton.appendChild(menuIcon);
+        overflowMenuButton.iconFixed = true;
```

Comment:
Grouping code by icon manipulated is easier to consume

```suggestion
        const menuIcon = document.createElement('vaadin-icon');
        menuIcon.setAttribute('icon', 'lumo:menu');
        menuIcon.classList.add('vaadin-menu-bar-button--icon');
        menuIcon.classList.add('menu-icon');
                        
        const closeIcon = document.createElement('vaadin-icon');
        closeIcon.setAttribute('icon', 'lumo:cross');
        closeIcon.classList.add('vaadin-menu-bar-button--icon');
        closeIcon.classList.add('close-icon');
        
        overflowMenuButton.appendChild(closeIcon);
        overflowMenuButton.appendChild(menuIcon);
        overflowMenuButton.iconFixed = true;
```